magefiles/magefile.go

```diff
@@ -331,6 +331,12 @@ func (t Test) UpdateVMGolden() error {
 	return sh.RunWithV(ENV, "go", "test", "-v", "-tags=vm_integration", "./integration/...", "-update")
 }
 
+// E2e runs E2E tests using testscript framework
+func (t Test) E2e() error {
+	mg.Deps(Install) // Install trivy to $GOPATH/bin
```

Comment:
Just a thought - should we keep a separate install path for this binary in order to not pollute the existing trivy $GOPATH/bin setup? Since we're using an existing directory - there's a small concern I have where we might be running a wrong Trivy artifact.

Reply:
You’re right. I had the same concern, so at first I tried running the pre-built binary directly through `mage build`, but since the testscript runs inside a temporary directory called `$WORK`, which is a unique path for each test, it brought problems with complex relative paths. It’s also possible to build the binary within each test, but that means rebuilding it for each test, and even if we reuse the module cache, the tests still become quite heavy. I couldn’t find a better solution, so in the end I gave up and resorted to using $GOPATH. 

I’ll keep thinking about a better approach, but if anyone comes up with a good idea, please let me know.